src/mpi/init/mpir_init.c

```diff
@@ -158,17 +158,19 @@ int MPIR_Init_thread_impl(int *argc, char ***argv, int user_required, int *provi
     /**********************************************************************/
 
     /* Setting isThreaded to 0 to trick any operations used within
-     * MPID_Init to think that we are running in a single threaded
+     * MPID_Init_local/world to think that we are running in a single threaded
      * environment. */
 #ifdef MPICH_IS_THREADED
     MPIR_ThreadInfo.isThreaded = 0;
 #endif
 
     MPL_atomic_store_int(&MPIR_Process.mpich_state, MPICH_MPI_STATE__IN_INIT);
 
-    mpi_errno = MPID_Init(required, &MPIR_ThreadInfo.thread_provided);
+    mpi_errno = MPID_Init_local(required, &MPIR_ThreadInfo.thread_provided);
     MPIR_ERR_CHECK(mpi_errno);
 
+    mpi_errno = MPID_Init_world();
+    MPIR_ERR_CHECK(mpi_errno);
 
```

Comment:
Is it OK to remove existing `MPID_` interface? It is not backward compatible.

Reply:
I see. That is a good point. Let me work out something to keep the backward compatibility.

Reply:
I removed this commit. I'll workout in the session PR to maintain the backward compatibility.